cli/exchange/node.go

```diff
@@ -465,3 +480,341 @@ func NodeListErrors(org string, credToUse string, node string, long bool) {
 		fmt.Printf("%s\n", jsonBytes)
 	}
 }
+
+func ValidateUserInput(org string, credToUse string, userInput policy.UserInput, exNode ExchangeNodes) (bool, string, error) {
+	msgPrinter := i18n.GetMessagePrinter()
+	warningMessage := ""
+
+	serviceOrg := userInput.ServiceOrgid
+	serviceUrl := userInput.ServiceUrl
+	serviceVersionRange := userInput.ServiceVersionRange
+	serviceArch := userInput.ServiceArch
+
+	msgPrinter.Printf("Validating userinput %v, %v, %v, %v", serviceOrg, serviceUrl, serviceVersionRange, serviceArch)
+	msgPrinter.Println()
+
+	var errorString string
+	if &userInput.ServiceVersionRange == nil || userInput.ServiceVersionRange == "" {
+		def := "0.0.0"
+		serviceVersionRange = def
+	}
+
+	// Convert the version to a version expression.
+	vExp, err := semanticversion.Version_Expression_Factory(serviceVersionRange)
+	if err != nil {
+		errorString = msgPrinter.Sprintf("versionRange %v cannot be converted to a version expression, error %v", userInput.ServiceVersionRange, err)
+		return false, warningMessage, errors.New(errorString)
+	}
+	serviceVersion := vExp.Get_expression()
+
+	// service exist
+	var sdef *exchange.ServiceDefinition
+
+	searchVersion, err := getSearchVersion(serviceVersion)
+	if err != nil {
+		return false, warningMessage, err
+	}
+
+	sdef, err = getService(cliutils.OrgAndCreds(org, credToUse), serviceOrg, serviceUrl, serviceArch, serviceVersion, searchVersion)
+	if sdef == nil || err != nil {
+		return false, warningMessage, err
+	}
+
+	// compare ServiceDefinition.UserInput(array) with userInput.Inputs(array)
+	var compareResult bool
+	compareResult, err = compareUserInput(sdef.UserInputs, userInput.Inputs)
+	if !compareResult {
+		errorString = msgPrinter.Sprintf("Failed to compare given userInput with service userInput, error: %v", err)
```

Comment:
Please change the wording of the message to:
errorString = msgPrinter.Sprintf("Failed comparing userInput with service's userInput, error: %v", err)